### PR TITLE
Improve protocol section

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Improve protocol section to give information about what is it and why is it improtant for new users.

Changes:
* index.md: removed the technical description and contract description and moved to the new "registry_technical_overview" section. Added more introductory information for the developer to explain the motivation of the Autonolas protocol.
* contract_addresses.md -> renamed to registry_technical_overview.md: Contains the former information from the index page (technical details and contracts of the registry) plus the contract addresses in a section in the bottom.
* Added a few images.

### CODE REVIEW

File: docs/protocol/index.md [hunk 1]
```diff
@@ -4,55 +4,17 @@
...
+Autonolas strives to incentivize software composability and reusability. Thus, the Autonolas Protocol reflects such composability by representing software components as NFTs, and incorporates a mechanism that enables combining them to take into account their actual software composition relationship. This is a crucial feature to measure their contribution and reward according to the protocol tokenomics model.
 
-The Autonolas Protocol is built with the {{open_autonomy}} framework in mind as the primary framework for realizing agent services. However, it does not prescribe the usage of the {{open_autonomy}} framework and allows for services to be implemented on alternative frameworks.
+The Autonolas Protocol is [currently deployed](./technical_overview.md#contract-addresses) in Ethereum and Görli chains, and it will be progressively deployed in all major blockchains.
 
 !!! abstract "Learn more"
 
     Read the **Technical Architecture**, **Tokenomics** and **Governance** sections in the [Autonolas Whitepaper](https://www.autonolas.network/documents/whitepaper/Whitepaper%20v1.0.pdf) for the full details of the **Autonolas Protocol**.
-
```
Comment (mariapiamo) [inline docs/protocol/index.md:25]: Do you think we can or have to cite the various repository and documentation therein?

File: docs/protocol/index.md [hunk 2]
```diff
@@ -4,55 +4,17 @@
 ![Autonolas Protocol logo](images/protocol.png){ align=left width="150" }
 </figure>
 
-The on-chain **Autonolas Protocol** anchors the Autonolas autonomous services -and in particular the current incarnation as agent services- on the target settlement layer and provides the primitives needed to create, operate and secure such services. The protocol benefits from a modular design with a clear separation of concerns and opportunity for extensibility without compromising its security and permissionless nature.
+The on-chain **Autonolas Protocol** anchors the Autonolas autonomous services -and in particular the current incarnation as agent services- on the target settlement layer and provides the primitives needed to create, operate and secure such services.
+The protocol also provides a mechanism that incentivizes their creation and **rewards developers and service operators proportionally for their efforts to support the growth of the Autonolas ecosystem**.
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply docs/protocol/index.md]: fixed

File: docs/protocol/index.md [hunk 3]
```diff
@@ -4,55 +4,17 @@
...
-
-* Follow a core-periphery architecture (such as in Uniswap), which allows for changing out periphery functionality without changing the data models at the core.
-* Allow for extension via modules (such as in MakerDAO).
+<figure markdown>
+![Package minting](./images/ecosystem_flywheel.svg)
+</figure>
 
-Examples of modules include governance and staking. Governance is particularly important in a modular system, as it is used to vote on the adoption or abandoning of modules. By ensuring an immutable core, the Autonolas protocol provides guarantees that once created, the ecosystem’s agent components, canonical agents, and services are not mutable by governance – an important guarantee of censorship resistance.
+Autonolas strives to incentivize software composability and reusability. Thus, the Autonolas Protocol reflects such composability by representing software components as NFTs, and incorporates a mechanism that enables combining them to take into account their actual software composition relationship. This is a crucial feature to measure their contribution and reward according to the protocol tokenomics model.
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply docs/protocol/index.md]: rewriten

File: docs/protocol/technical_overview.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+Examples of modules include governance and staking. Governance is particularly important in a modular system, as it is used to vote on the adoption or abandoning of modules. By ensuring an immutable core, the Autonolas protocol provides guarantees that once created, the ecosystem’s agent components, canonical agents, and services are not mutable by governance – an important guarantee of censorship resistance.
+
+The Autonolas Protocol is built with the {{open_autonomy}} framework in mind as the primary framework for realizing agent services. However, it does not enforces the usage of the {{open_autonomy}} framework, and it allows for services to be implemented on alternative frameworks.
+
+!!! abstract "Learn more"
+
+    Read the **Technical Architecture**, **Tokenomics** and **Governance** sections in the [Autonolas Whitepaper](https://www.autonolas.network/documents/whitepaper/Whitepaper%20v1.0.pdf) for the full details of the **Autonolas Protocol**.
+
+## Core smart contracts
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply docs/protocol/technical_overview.md:14]: fixed.

**Comment by jmoreira-valory** [comment]
Comments/Ideas:

1. Currently we only detail de registry. Should we detail as well the Tokenomics and Governance? (We are giving pointers in the main page).
2. Missing a toy example with some numbers, etc. of a developer that creates a component and his rewards are computed?
3. More in depth in any section? e.g. reward measure?

File: docs/protocol/index.md [hunk 5]
```diff
@@ -4,55 +4,36 @@
 ![Autonolas Protocol logo](images/protocol.png){ align=left width="150" }
 </figure>
 
-The on-chain **Autonolas Protocol** anchors the Autonolas autonomous services -and in particular the current incarnation as agent services- on the target settlement layer and provides the primitives needed to create, operate and secure such services. The protocol benefits from a modular design with a clear separation of concerns and opportunity for extensibility without compromising its security and permissionless nature.
+The **Autonolas Protocol** is a collection of smart contracts that implements the mechanism to secure open-source code on-chain, and provides incentives to developers proportionally for their efforts in supporting the growth of the Autonolas ecosystem. The protocol is built with the {{open_autonomy}} framework in mind as the primary framework for realizing agent services,  alternative frameworks can also be used.
 
-!!! info
+The Autonolas Protocol is [currently deployed](./registry_technical_overview.md#contract-addresses) in Ethereum mainnet and Görli testnet, and it will be eventually deployed in all major smart-contract blockchains.
```
Comment (oaksprout) [inline docs/protocol/index.md:9]: in Ethereum -> on Ethereum

Comment (oaksprout) [reply docs/protocol/index.md:9]: in all -> on all

File: docs/protocol/index.md [hunk 6]
```diff
@@ -4,55 +4,36 @@
...
 
-From an architectural point of view, the smart contracts that make up the Autonolas Protocol satisfy the following properties:
+There are three main elements that make up the protocol:
 
-* Follow a core-periphery architecture (such as in Uniswap), which allows for changing out periphery functionality without changing the data models at the core.
-* Allow for extension via modules (such as in MakerDAO).
+* The **On-Chain Registry** anchors Autonolas autonomous services on-chain, and provides the primitives needed to create, operate and secure such services.
 
-Examples of modules include governance and staking. Governance is particularly important in a modular system, as it is used to vote on the adoption or abandoning of modules. By ensuring an immutable core, the Autonolas protocol provides guarantees that once created, the ecosystem’s agent components, canonical agents, and services are not mutable by governance – an important guarantee of censorship resistance.
+* The **Tokenomics** defines economic model facilitated by the Autonolas token. This is a mechanism to grow the capital deployed in services and reward developers.
```
Comment (oaksprout) [inline docs/protocol/index.md:17]: defines an economic

File: docs/protocol/index.md [hunk 7]
```diff
@@ -4,55 +4,36 @@
...
 
-The Autonolas Protocol is built with the {{open_autonomy}} framework in mind as the primary framework for realizing agent services. However, it does not prescribe the usage of the {{open_autonomy}} framework and allows for services to be implemented on alternative frameworks.
+* The **Governance** defines the rules of Autonolas as a decentralized, autonomous
```
Comment (oaksprout) [inline docs/protocol/index.md:19]: The Governance -> Governance

File: docs/protocol/index.md [hunk 8]
```diff
@@ -4,55 +4,36 @@
...
-**Unit Registry**
-:	An abstract smart contract for generic agents/components template which inherits the Generic Registry
-
-**Component Registry**
-:	An ERC721 contract that inherits the Unit Registry and represents agent components.
+## Rationale
 
-**Agent Registry**
-:	An ERC721 contract that inherits the Unit Registry and represents canonical agents.
+In most settings, the reward model for service owners and agent operators is usually straightforward: users remunerate service owners, and service owners remunerate operators supporting their service. However it is not always well-defined how this is propagated to open software developers. This is where the Autonolas Protocol plays its role.
```
Comment (oaksprout) [inline docs/protocol/index.md:32]: I guess this should be "open-source software developers" rather than "open software"?

**Review by oaksprout — CHANGES_REQUESTED** [review]

File: docs/protocol/index.md [hunk 9]
```diff
@@ -4,55 +4,35 @@
...
+There are three main elements that make up the protocol:
 
-* Follow a core-periphery architecture (such as in Uniswap), which allows for changing out periphery functionality without changing the data models at the core.
-* Allow for extension via modules (such as in MakerDAO).
+* The **On-Chain Registry** anchors Autonolas autonomous services on-chain, and provides the primitives needed to create, operate and secure such services.
 
-Examples of modules include governance and staking. Governance is particularly important in a modular system, as it is used to vote on the adoption or abandoning of modules. By ensuring an immutable core, the Autonolas protocol provides guarantees that once created, the ecosystem’s agent components, canonical agents, and services are not mutable by governance – an important guarantee of censorship resistance.
+* The **Tokenomics** defines an economic model facilitated by the [OLAS token](https://etherscan.io/token/0x0001A500A6B18995B03f44bb040A5fFc28E45CB0). This is a mechanism to grow the capital deployed in services and reward developers.
```
Comment (mariapiamo) [inline docs/protocol/index.md:17]:
```suggestion
* The **Tokenomics** defines an economic model that uses the [OLAS token](https://etherscan.io/token/0x0001A500A6B18995B03f44bb040A5fFc28E45CB0) as a coordination mechanism to accomplish three main objectives:
- enable the pairing of capital and code in a permissionless manner
-  create a flywheel that attracts increasingly more value
and provides truly-decentralized autonomous services, owned by a DAO, operated by ecosystem actors, and coded by the ecosystem developers
- incentivize software composability.
```

File: docs/protocol/index.md [hunk 10]
```diff
@@ -4,55 +4,35 @@
...
 
-The Autonolas Protocol is built with the {{open_autonomy}} framework in mind as the primary framework for realizing agent services. However, it does not prescribe the usage of the {{open_autonomy}} framework and allows for services to be implemented on alternative frameworks.
+* **Governance** defines the rules of Autonolas as a decentralized, autonomous
```
Comment (mariapiamo) [inline docs/protocol/index.md:19]:
```suggestion
* **Governance**  allows the Autonolas DAO as a decentralized, autonomous
```

File: docs/protocol/index.md [hunk 11]
```diff
@@ -4,55 +4,35 @@
...
+organization (DAO), which is governed by the community.
```
Comment (mariapiamo) [inline docs/protocol/index.md:20]:
```suggestion
organization (DAO), to steer and fine-tune aspects of the Autonola protocol overtime.
```

**Comment by jmoreira-valory** [comment]
@mariapiamo should we remove the instances of open-source software in the text ?

File: docs/protocol/index.md (hunk 1 above)
Comment (jmoreira-valory) [reply docs/protocol/index.md:25]: Sorry, I didn't make my point clear. 

I meant, as "owner" of this info textbox (which I also agree is useful here), do you find approppriate to insert the links to the details and technical repos (as suggested above by @mariapiamo) in this short text ?

Comment (oaksprout) [reply docs/protocol/index.md:25]: Ah I see, no that doesn't seem necessary to me

**Comment by mariapiamo** [comment]
> @mariapiamo should we remove the instances of open-source software in the text?

just in the point as suggested it. The other parts look good

**Review by mariapiamo — APPROVED** [review]

**Review by dagacha — COMMENTED** [review]
Made a few changes to improve the text. Please check @jmoreira-valory

File: docs/protocol/index.md [hunk 12]
```diff
@@ -4,55 +4,40 @@
...
 
-**Registries Manager**
-:	A contract inheriting from Generic Manager via which developers can mint component and agent NFTs.
+Autonolas proposes a model where open-source developers, that contribute to the community, benefit from those contributions. This model consists on incentivizing software **composability**, **reusability** and **utility**. Roughly speaking, software packages (components and agents) brought to the Autonolas ecosystem are secured and minted as NFTs in the Autonolas protocol. These packages can be used to code agents and services by composition, and the protocol has mechanisms to unambiguously represent the actual software/system composition on-chain. This is a crucial feature to [measure the utility](https://github.com/valory-xyz/autonolas-tokenomics/blob/main/docs/Autonolas_tokenomics_audit.pdf) of the code brought by developers and provide a fair reward for their contributions.
```
Comment (jmoreira-valory) [inline docs/protocol/index.md:39]:
```suggestion
Autonolas proposes a model where open-source developers which contribute to the community benefit from those contributions. This model consists on incentivizing software **composability**, **reusability** and **utility**. Roughly speaking, software packages (components and agents) brought to the Autonolas ecosystem are secured and minted as NFTs in the Autonolas protocol. These packages can be used to code agents and services by composition, and the protocol has mechanisms to unambiguously represent the actual software/system composition on-chain. This is a crucial feature to [measure the utility](https://github.com/valory-xyz/autonolas-tokenomics/blob/main/docs/Autonolas_tokenomics_audit.pdf) of the code brought by developers and provide a fair reward for their contributions.
```

**Review by dagacha — APPROVED** [review]

**Pull request merged**